ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/datalayer/datalayer.js

```diff
@@ -87,39 +95,61 @@
                 }
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: getEventInfo(FormEvents.FIELD, formTitle, event.detail.fieldId, dataLayerContent)
+                    eventInfo: getEventInfo(FormEvents.FIELD, event.detail.formTitle, event.detail.fieldId, dataLayerContent, formContainerID)
                 });
             });
 
             bridge.on('elementHelpShown', function(event){
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: getEventInfo(FormEvents.HELP, event.detail.formTitle, event.detail.fieldId, dataLayerContent)
+                    eventInfo: getEventInfo(FormEvents.HELP, event.detail.formTitle, event.detail.fieldId, dataLayerContent, formContainerID)
                 });
             });
 
             bridge.on('elementErrorShown', function(event){
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: getEventInfo(FormEvents.ERROR, event.detail.formTitle, event.detail.fieldId, dataLayerContent)
+                    eventInfo: getEventInfo(FormEvents.ERROR, event.detail.formTitle, event.detail.fieldId, dataLayerContent, formContainerID)
                 });
             });
 
             bridge.on('submitStart', function(event){
-                let eventInfo = {
```

Comment:
pls break all anonymous event handlers in separate functions for readability. they should be outside the scope of the parent function.

Reply:
done